views/js/apipCreator/api/accessElement.js

```diff
@@ -0,0 +1,233 @@
+    'lodash',
+    'taoQtiItem/apipCreator/api/authoringObject',
+    'taoQtiItem/apipCreator/api/qtiElement',
+    'taoQtiItem/apipCreator/api/accessElementInfo'
+], function (_, authoringObject, QtiElement, AccessElementInfo) {
+    'use strict';
+    
+    /**
+     * accessElement constructor
+     * @param {object} apipItem apipItem creator api instance 
+     * @see {@link package-tao\taoQtiItem\views\js\apipCreator\api\apipItem.js}
+     * @param {type} node XML element node. If not given then new empty node will be created and appended to the item XML.
+     * @returns {accessElement_L19.AccessElement}
+     */
+    function AccessElement(apipItem, node) {
+        if (node === undefined) {
+            node = this.createXMLNode(apipItem);
+        }
+        authoringObject.init(this, apipItem, node);
+    }
+
+    /**
+     * Create accessElement xml node and append it to the item XML
+     * @param {object} apipItem apipItem creator api instance 
+     * @see {@link package-tao\taoQtiItem\views\js\apipCreator\api\apipItem.js}
+     * @returns {Object} created XML node
+     */
+    AccessElement.prototype.createXMLNode = function createXMLNode(apipItem) {
+        var accessElementNum = apipItem.xpath('//apip:accessElement').length,
+            accessElementNode = apipItem.createNode(
+                'apip',
+                'accessElement',
+                {"identifier" : ('ae' + accessElementNum)}
+            ),
+            accessElementInfoNode = apipItem.createNode('apip', 'relatedElementInfo');
+
+        accessElementNode.appendChild(accessElementInfoNode);
+        apipItem.xpath('//apip:accessibilityInfo')[0].appendChild(accessElementNode);
+        return accessElementNode;
+    };
+
+    /**
+     * Find the related QtiElements array from the associated accessElement
+     * 
+     * @returns {array}
+     */
+    AccessElement.prototype.getQtiElements = function getQtiElements() {
+        var that = this,
+            contentLinks = this.apipItem.xpath("//apip:accessElement[@serial='" + that.serial + "']/apip:contentLinkInfo"),
+            result = [];
+
+        _.forEach(contentLinks, function (link) {
+            var node = that.apipItem.xpath("qti:itemBody//*[@id='" + link.getAttribute('qtiLinkIdentifierRef') + "']");
+
+            if (node && node.length) {
+                result.push(new QtiElement(this, node[0]));
+            }
+        });
+
+        return result;
+    };
+
+    /**
+     * Add related QtiElement to the accessElement (add <apip:contentLinkInfo> element)
+     * 
+     * @param {object} qtiElement QtiElement instance
+     * @returns {Object} ceated <apip:contentLinkInfo> node
+     */
+    AccessElement.prototype.addQtiElement = function addQtiElement(qtiElement) {
+        var that = this,
+            contentLinkInfo = that.apipItem.createNode(
+                'apip',
+                'contentLinkInfo',
+                {"qtiLinkIdentifierRef" : qtiElement.data.getAttribute('id')}
```

Comment:
the attribute id on an qtiElement is required to link it to an accessElement . However it is not mendatory in the qti standard. So should we not check if the attribute id exists for the qtiElement or generate an unique one for the qtiElement here or there https://github.com/oat-sa/extension-tao-itemqti/blob/TAO-795_APIP_authoring_api/views/js/apipCreator/api/qtiElement.js#L79 ?